keras_cv/models/object_detection/yolo_v8/yolo_v8_label_encoder.py

```diff
@@ -0,0 +1,287 @@
+            mask_multi_gts, is_max_overlaps, mask_pos
+        )  # (b, max_num_boxes, num_anchors)
+        fg_mask = tf.reduce_sum(mask_pos, axis=-2)
+        return fg_mask, mask_pos
+
+    fg_mask, mask_pos = tf.cond(
+        tf.reduce_max(fg_mask) > 1,
+        lambda: handle_anchor_with_two_gt_boxes(
+            fg_mask, mask_pos, overlaps, max_num_boxes
+        ),
+        lambda: (fg_mask, mask_pos),
+    )
+
+    target_gt_idx = tf.argmax(mask_pos, axis=-2)  # (b, num_anchors)
+    return target_gt_idx, fg_mask, mask_pos
+
+
+def select_candidates_in_gts(xy_centers, gt_bboxes, epsilon=1e-9):
+    """Selects candidate anchors for GT boxes"""
+    n_anchors = xy_centers.shape[0]
+    bs, n_boxes, _ = gt_bboxes.shape
+
+    lt, rb = tf.split(
+        tf.reshape(gt_bboxes, (-1, 1, 4)), 2, axis=-1
+    )  # left-top, right-bottom
+    bbox_deltas = tf.reshape(
+        tf.concat(
+            [xy_centers[tf.newaxis] - lt, rb - xy_centers[tf.newaxis]], axis=2
+        ),
+        (-1, n_boxes, n_anchors, 4),
+    )
+
+    return tf.reduce_min(bbox_deltas, axis=-1) > epsilon
+
+
+@keras.utils.register_keras_serializable(package="keras_cv")
+class YOLOV8LabelEncoder(layers.Layer):
+    def __init__(
+        self, num_classes, topk=10, alpha=0.5, beta=6.0, epsilon=1e-9, **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.topk = topk
+        self.num_classes = num_classes
+        self.alpha = alpha
+        self.beta = beta
+        self.epsilon = epsilon
+
+    def call(
+        self, pd_scores, pd_bboxes, anc_points, gt_labels, gt_bboxes, mask_gt
+    ):
+        max_num_boxes = gt_bboxes.shape[1]
+
+        mask_pos, align_metric, overlaps = self.get_pos_mask(
+            pd_scores,
+            pd_bboxes,
+            gt_labels,
+            gt_bboxes,
+            anc_points,
+            mask_gt,
+            max_num_boxes,
+        )
+
+        target_gt_idx, fg_mask, mask_pos = select_highest_overlaps(
+            mask_pos, overlaps, max_num_boxes
+        )
+
+        # assigned target
+        target_labels, target_bboxes, target_scores = self.get_targets(
+            gt_labels, gt_bboxes, target_gt_idx, fg_mask, max_num_boxes
+        )
+
+        # normalize
+        align_metric *= mask_pos
+        pos_align_metrics = tf.reduce_max(
+            align_metric, axis=-1, keepdims=True
+        )  # b, max_num_boxes
+        pos_overlaps = tf.reduce_max(
+            overlaps * mask_pos, axis=-1, keepdims=True
+        )  # b, max_num_boxes
+        norm_align_metric = tf.expand_dims(
+            tf.reduce_max(
+                align_metric
+                * pos_overlaps
+                / (pos_align_metrics + self.epsilon),
+                axis=-2,
+            ),
+            axis=-1,
+        )
+        target_scores = target_scores * norm_align_metric
+
+        # No need to compute gradients for these, as they're all targets
+        return (
+            tf.stop_gradient(target_bboxes),
```

Comment:
Maybe we should do this in the RetinaNet for performance as well.

Reply:
I didn't a/b test the performance, but I figured it was nice to include at the very least from a readability perspective.